plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TableSnapshot.java

```diff
@@ -71,7 +72,9 @@ private TableSnapshot(
             String tableLocation,
             ParquetReaderOptions parquetReaderOptions,
             boolean checkpointRowStatisticsWritingEnabled,
-            int domainCompactionThreshold)
+            int domainCompactionThreshold,
+            Optional<MetadataEntry> cachedMetadata,
+            Optional<ProtocolEntry> cachedProtocol)
```

Comment:
From the code changes i see that the metadata & protocol are always obtained from the `logTail`.
Why do we add then those two parameters to the constructor?

Reply:
We try to get them from the log tail every time, but most of the time they'll be empty, so we need the cached version somehow.